### PR TITLE
Add support for charts to the extension API

### LINKED ISSUE #10018: Prototype graph contributions
Part of [RFC 144: Code Insights](https://docs.google.com/document/d/1EHzor6I1GhVVIpl70mH-c10b1tNEl_p1xRMJ9qHQfoc/edit#heading=h.trqab8y0kufp)

### PR DESCRIPTION
**Pull request by felixfbecker**
Adds support for charts in the extension API. It basically exposes a declarative, object literal API over recharts. I'm not 100% sure about all API aspects but am happy enough with this version to ship it for experimentation 🚢 

Example usages: 
- Line charts: https://github.com/sourcegraph/sourcegraph-search-insights
- Pie charts: https://github.com/codecov/sourcegraph-codecov/pull/52

Testing strategy: Storybooks.

Closes #10018

### CODE REVIEW

**Comment by lguychard** [comment]
@twop this would be interesting for you to review 🙂

File: web/src/views/niceTicks.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
+// Copied from https://github.com/cenfun/nice-ticks/blob/f310f77280fae72706f8b1f956d01029f0c4f133/src/index.js
```
Comment (eseliger) [inline web/src/views/niceTicks.ts:1]: maybe stuff that we need to copy in can all go to the same folder (rxjs fetch is also imported), and that folder can be excluded from code coverage? I think we can assume it is tracked outside of our repo

Comment (eseliger) [reply web/src/views/niceTicks.ts:1]: Also, how can we make this appear in the licenses file?

Comment (felixfbecker) [reply web/src/views/niceTicks.ts]: Used a fork instead 😄 

In general I'd prefer to not encourage vendoring.

Also dependency code should be subject to coverage requirements too - I'd actually love if Codecov could include coverage of dependencies into our coverage metrics.

Comment (felixfbecker) [reply web/src/views/niceTicks.ts]: Upstreamed the change!

Comment (eseliger) [reply web/src/views/niceTicks.ts]: nicé !

File: packages/sourcegraph-extension-api/src/sourcegraph.d.ts [hunk 2]
```diff
@@ -736,6 +830,7 @@ declare module 'sourcegraph' {
          */
         content: (
             | MarkupContent
+            | ChartContent
```
Comment (twop) [inline packages/sourcegraph-extension-api/src/sourcegraph.d.ts:833]: I like the simplicity of it 👍 

But overall I have a fear that this is yet one more "one of" cases for UI elements.
Example: I need to add a title content to a chart. Aka composability aspect of the API.

I don't think we need to address this concern right now but wanted to bring this up.

**Pull request merged**